### PR TITLE
Tools: allow to retry to get parameters before timeout

### PR DESCRIPTION
**Pull request by khancyr**
it should cope with _ENABLE parameter more easily

### CODE REVIEW

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -480,7 +481,7 @@ def autodisarm_motors(self):
         return False
 
     def set_parameter(self, name, value, add_to_context=True):
-        old_value = self.get_parameter(name)
+        old_value = self.get_parameter(name, num_rety=2)
```
Comment (peterbarker) [inline Tools/autotest/common.py]: `s/num_rety/retries/`

File: Tools/autotest/common.py [hunk 2]
```diff
@@ -493,10 +494,17 @@ def set_parameter(self, name, value, add_to_context=True):
...
+    def get_parameter(self, name, num_rety=1, timeout=60):
+        for i in range(0, num_rety):
+            self.mavproxy.send("param fetch %s\n" % name)
+            try:
+                self.mavproxy.expect("%s = ([-0-9.]*)\r\n" % (name,), timeout=timeout/num_rety)
+                return float(self.mavproxy.match.group(1))
+            except pexpect.TIMEOUT:
+                if i < num_rety:
+                    continue
+                else:
```
Comment (peterbarker) [inline Tools/autotest/common.py]: pointless else

**Comment by khancyr** [comment]
from my tests, self.mavproxy.expect isn't really reliable. if you look at travis log at https://github.com/ArduPilot/ardupilot/pull/9199 . It try to fetch GRIP_GRAB but failed when it should have success without problem.
I add the retry mechanism to not just wast the timeout time as it won't unlock on its one, giving a retry chance make the fetch pass !

File: Tools/autotest/common.py (hunk 1 above)
Comment (khancyr) [reply Tools/autotest/common.py]: done

File: Tools/autotest/common.py (hunk 2 above)
Comment (khancyr) [reply Tools/autotest/common.py]: hum so we don't stop the test when we cannot get the parameter ?

**Pull request merged**